pkg/nvlib/device/device.go

```diff
@@ -91,8 +91,6 @@ func (d *device) GetArchitectureAsString() (string, error) {
 		return "Hopper", nil
 	case nvml.DEVICE_ARCH_BLACKWELL:
 		return "Blackwell", nil
-	case nvml.DEVICE_ARCH_T23X:
```

Comment:
This is required since the `DEVICE_ARCH_T23X` definition was removed from `nvml.h` as part of the CUDA 13.0 release.